nmostesting/suites/BCP0060102Test.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (C) 2022 Advanced Media Workflow Association
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ..ControllerTest import ControllerTest, TestingFacadeException
+from ..NMOSUtils import NMOSUtils
+
+
+class BCP0060102Test(ControllerTest):
+    """
+    Runs Controller Tests covering BCP-006-01
+    """
+    def __init__(self, apis, registries, node, dns_server):
+        ControllerTest.__init__(self, apis, registries, node, dns_server)
+
+    def _generate_caps(self, media_types):
+        caps = {
+            "media_types": media_types, "constraint_sets": [
+                {
+                    "urn:x-nmos:cap:format:color_sampling": {
+                        "enum": ["YCbCr-4:2:2"]
+                    },
+                    "urn:x-nmos:cap:format:frame_height": {
+                        "enum": [1080]
+                    },
+                    "urn:x-nmos:cap:format:frame_width": {
+                        "enum": [1920]
+                    },
+                    "urn:x-nmos:cap:format:grain_rate": {
+                        "enum": [{
+                                "denominator": 1,
+                                "numerator": 25
+                        }]
+                    },
+                    "urn:x-nmos:cap:format:interlace_mode": {
+                        "enum": [
+                            "interlaced_bff",
+                            "interlaced_tff",
+                            "interlaced_psf"
+                        ]
+                    }
+                }
+            ],
+            "version": NMOSUtils.get_TAI_time()
+        }
+
+        return caps
+
+    def set_up_tests(self):
+        self.receivers = [{'label': 'r1/john', 'description': 'Mock receiver 1',
+                           'connectable': True, 'registered': True},
+                          {'label': 'r2/paul', 'description': 'Mock receiver 2',
+                           'connectable': True, 'registered': True},
+                          {'label': 'r3/george', 'description': 'Mock receiver 3',
+                           'connectable': True, 'registered': True},
+                          {'label': 'r4/ringo', 'description': 'Mock receiver 4',
+                           'connectable': True, 'registered': True}]
+
+        # Randomly select some Receivers to be JPEG XS capable
+        jxsv_receivers = self._generate_random_indices(len(self.receivers), min_index_count=1)
```

Comment:
The reasoning for an explicit count rather than a proportion is for the case of testing pagination where the test specifies 2 results per page, and a minimum of 3 to force more than one page.  Proportional percentages would be cumbersome in that context, I feel.

Reviewing the use of this function in the IS-04 and IS-05 tests, I'm not sure the max_index_count is needed - I will experiment with removing it.  Also the min_index_count should default to 1, and then be overridden in the cases where a minimum needs to be specified to force pagination.